src/cmd4.cc

```diff
@@ -181,7 +181,7 @@ void do_cmd_change_name(void)
 		else if (c == 'f')
 		{
 			strnfmt(tmp, 160, "%s.txt", player_name);
-			if (get_string("Filename(you can post it to http://angband.oook.cz/): ", tmp, 80))
+			if (get_filename("Filename(you can post it to http://angband.oook.cz/): ", tmp, 80))
```

Comment:
Yes, essentially that. I think there is at least one case where a path like "~/foo/..." is provided as a default, but that could obviously be 'fixed' in some way. I can't recall offhand where that would be. Not sure if it's a critical feature -- it might be annoying to have to save char dumps to ~/.tome/... since they're usually meant for uploading to the ladder.

BTW, I'm not necessarily **opposed** to restricting the file paths used, but I think it may need to be configurable somehow. (Perhaps a command line switch or perhaps a compile-time switch -- much as I loathe them?). I'd probably also be easier to convince if this was part of a general refactoring to start using boost::fs::path instead of plain strings :).

Obviously general security fixes such as fixing overflows are most welcome... and **these** issues are going to be the biggest ones by far when running T2 on a server with untrusted user inputs. Even though I've been running with sanitizer I'm sure there are still a *lot* of overflows all over the place.